native/explorer/src/series.rs

```diff
@@ -26,9 +26,30 @@ macro_rules! from_list {
 from_list!(s_from_list_i64, i64);
 from_list!(s_from_list_u32, u32);
 from_list!(s_from_list_bool, bool);
-from_list!(s_from_list_f64, f64);
 from_list!(s_from_list_str, String);
 
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn s_from_list_f64(name: &str, val: Term) -> ExSeries {
+    ExSeries::new(Series::new(
+        name,
+        val.decode::<ListIterator>()
+            .unwrap()
+            .map(|item| match item.get_type() {
+                TermType::Number => Some(item.decode::<f64>().unwrap()),
+                TermType::Atom => {
+                    let atom = item.atom_to_string().unwrap();
+                    if atom.contains("nil") {
+                        None
+                    } else {
+                        Some(cast_str_to_f64(atom.as_str()))
+                    }
```

Comment:
Since we have a limited list of atoms that we want to compare, I think we could create those atoms at the top of this funcion, and compare them with https://docs.rs/rustler/latest/rustler/types/atom/struct.Atom.html#method.eq-1

This would reduce the number of allocations, since you would avoid creating a string each time you see the atom.
The only drawback I see is that you would need to check each value, but I think this is fine. WDYT?

Reply:
It makes sense! (https://github.com/elixir-nx/explorer/pull/495/commits/93b64a2fa0719d9cc18da11e822fcbed00d24698)